crates/core/tedge_agent/src/agent.rs

```diff
@@ -183,6 +184,7 @@ impl SmAgentConfig {
             .with_port(tedge_config.query(MqttPortSetting)?.into())
             .with_max_packet_size(10 * 1024 * 1024)
             .with_session_name(TEDGE_AGENT)
+            .with_initial_message(|| health_status_up_message("tedge-agent"))
             .with_last_will_message(health_status_down_message(TEDGE_AGENT));
```

Comment:
Use the daemon name when up and down.

```suggestion
            .with_initial_message(|| health_status_up_message(TEDGE_AGENT))
            .with_last_will_message(health_status_down_message(TEDGE_AGENT));
```